Src/StackifyLib/Internal/Auth/Claims/AppClaimsBuilderFullFramework.cs

```diff
@@ -187,12 +187,21 @@ private void SetWindowServiceAppName()
             catch (Exception ex)
             {
                 StackifyAPILogger.Log("Unable to get windows service name\r\n" + ex.ToString(), true);
-            }            
+            }
         }
 
         private async Task SetDeviceName()
         {
-            AppClaims.DeviceName = await GetEC2InstanceId() ?? Environment.MachineName;
+            var machineName = Environment.MachineName;
+
+            if (Config.IsEc2 == null || Config.IsEc2 == true || (machineName.StartsWith("EC2") && machineName.Contains("-")))
```

Comment:
for the sake of de-duping lines of code, what do you think about moving this code into the `AppClaimsBuilderBase`? something like:
- abstract method `AppClaimsBuilderBase.GetMachineName()`  that uses the framework specific calls
- `SetDeviceName()` becomes a protected method on the base class that calls `base.GetMachineName()` and then executes the EC2 code
- `GetEC2InstanceId()` becomes a abstract method on the base class as well